contracts/src/TeleporterTokenSource.sol

```diff
@@ -181,44 +181,75 @@ abstract contract TeleporterTokenSource is
      * - {input.recipient} cannot be the zero address
      * - {amount} must be greater than 0
      */
-    function _send(
+    function _send(SendTokensInput memory input, uint256 amount) internal sendNonReentrant {
+        _validateSendTokensInput(input);
+        // Require that a single hop transfer does not have a multi-hop fallback recipient.
+        require(
+            input.multiHopFallback == address(0),
+            "TeleporterTokenSource: non-zero multi-hop fallback"
+        );
```

Comment:
It may be worth putting this in a `_validateSingleHopInput` to be consistent with `TeleporterTokenDestination.sol`.

Reply:
I think the added function call may add a bit of gas usage, since we're passing a `memory` param here, without adding any extra readability

Reply:
Fair enough. It always irks me that there's tradeoff between readability/consistency and gas usage in Solidity, but c'est la vie.